modules/consensus/accept_test.go

```diff
@@ -701,3 +701,66 @@ func TestTaxHardfork(t *testing.T) {
 		t.Fatal("siafund pool was not increased correctly")
 	}
 }
+
+// mockGatewayDoesBroadcast implements modules.Gateway to mock the Broadcast
+// method.
+type mockGatewayDoesBroadcast struct {
+	modules.Gateway
+	broadcastCalled chan struct{}
+}
+
+// Broadcast is a mock implementation of modules.Gateway.Broadcast that
+// sends a sentinel value down a channel to signal it's been called.
+func (g *mockGatewayDoesBroadcast) Broadcast(string, interface{}) {
+	g.broadcastCalled <- struct{}{}
+}
+
+// TestAcceptBlockBroadcasts tests that AcceptBlock broadcasts valid blocks and
+// that managedAcceptBlock does not.
+func TestAcceptBlockBroadcasts(t *testing.T) {
+	cst, err := blankConsensusSetTester("TestAcceptBlockBroadcasts")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst.Close()
+	mg := &mockGatewayDoesBroadcast{
+		Gateway:         cst.cs.gateway,
+		broadcastCalled: make(chan struct{}),
+	}
+	cst.cs.gateway = mg
```

Comment:
Good point. Although in this test we only want to mock the gateway for the consensus set, and we don't want our mocked gateway to be used by transaction pool. If transaction pool were to call Broadcast we might mistake it as consensus calling Broadcast.